sycl/include/CL/sycl/detail/buffer_impl.hpp

```diff
@@ -143,9 +143,18 @@ class buffer_impl final : public SYCLMemObjT<AllocatorT> {
   }
 
   void *allocateMem(ContextImplPtr Context, bool InitFromUserData,
-                    RT::PiEvent &OutEventToWait) override {
+                    void *HostPtr, RT::PiEvent &OutEventToWait) override {
 
-    void *UserPtr = InitFromUserData ? BaseT::getUserPtr() : nullptr;
+    assert((!InitFromUserData || !static_cast<bool>(HostPtr)) &&
```

Comment:
`HostPtr == nullptr` seems less cryptic to me.

Reply:
Improved.